fiftyone/server/routes/embeddings.py

```diff
@@ -30,7 +36,7 @@ async def post(self, request: Request, data: dict) -> dict:
         stages = data["view"]
         label_field = data["labelField"]
 
-        dataset = _load_dataset(dataset_name)
+        dataset = fosu.load_and_cache_dataset(dataset_name)
```

Comment:
Since datasets with embeddings are explicitly cached, won't we get an error if we delete a dataset after plotting it and recreate a dataset with the same name as before?

Reply:
Well based on you and I's earlier experiments, what may previously happen in that case is that `load_dataset()` would return an instance with `name=None`. Hence the other change in this PR: now `load_dataset()` should detect that case and go ahead and reload the dataset from mongo, which is what we want.